The introduced change to `scripts/west_commands/tests/test_dfu_util.py` was necessary for fixing the following pytest errors under Python 3.12:

```
========================================================================================= short test summary info ==========================================================================================

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=dfu-util,alt=1,dfuse_config=None,img=test-zephyr.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=dfu-util,alt=alt-name,dfuse_config=None,img=test-zephyr.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=test-dfu-util,alt=1,dfuse_config=None,img=test-zephyr.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=dfu-util,alt=1,dfuse_config=None,img=test-img.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=dfu-util,alt=1,dfuse_config=DfuSeConfig(address=2, options='test-dfuse-opt'),img=test-zephyr.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_init[exe=dfu-util,alt=1,dfuse_config=DfuSeConfig(address=2, options=''),img=test-zephyr.bin] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=None,alt=1,dfuse_config=False,img=None0] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=None,alt=alt-name,dfuse_config=False,img=None] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=test-dfu-util,alt=1,dfuse_config=False,img=None] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=None,alt=1,dfuse_config=False,img=None1] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=None,alt=1,dfuse_config=True,img=test-dfuse-opt] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

FAILED tests/test_dfu_util.py::test_dfu_util_create[exe=None,alt=1,dfuse_config=True,img=None] - AttributeError: 'called_with' is not a valid assertion. Use a spec for the mock if 'called_with' is meant to be an attribute.

================================================================================ 12 failed, 536 passed, 4 warnings in 6.56s ================================================================================
```

--- a/.github/workflows/devicetree_checks.yml
+++ b/.github/workflows/devicetree_checks.yml
@@ -26,7 +26,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-22.04, macos-11, windows-2022]
         exclude:
         - os: macos-11

--- a/.github/workflows/pylib_tests.yml
+++ b/.github/workflows/pylib_tests.yml
@@ -25,7 +25,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-22.04]
     steps:
     - name: checkout

--- a/.github/workflows/scripts_tests.yml
+++ b/.github/workflows/scripts_tests.yml
@@ -25,7 +25,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-20.04]
     steps:
       - name: checkout

--- a/.github/workflows/twister_tests.yml
+++ b/.github/workflows/twister_tests.yml
@@ -29,7 +29,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-22.04]
     steps:
     - name: checkout

--- a/.github/workflows/twister_tests_blackbox.yml
+++ b/.github/workflows/twister_tests_blackbox.yml
@@ -19,7 +19,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-22.04]
     container:
       image: ghcr.io/zephyrproject-rtos/ci:v0.26.5

--- a/.github/workflows/west_cmds.yml
+++ b/.github/workflows/west_cmds.yml
@@ -27,7 +27,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         os: [ubuntu-22.04, macos-11, windows-2022]
         exclude:
         - os: macos-11

--- a/scripts/west_commands/tests/test_dfu_util.py
+++ b/scripts/west_commands/tests/test_dfu_util.py
@@ -86,7 +86,7 @@ def test_dfu_util_init(cc, req, find_device, tc, runner_config):
     with patch('os.path.isfile', side_effect=os_path_isfile_patch):
         runner.run('flash')
     assert find_device.called
-    assert req.called_with(exe)
+    assert req.call_args_list == [call(exe)]
     assert cc.call_args_list == [call(EXPECTED_COMMAND[tc])]
 
 def get_flash_address_patch(args, bcfg):
@@ -141,5 +141,5 @@ def test_dfu_util_create(cc, req, gfa, find_device, tc, runner_config, tmpdir):
         cfg = None
     map_tc = (exe or DFU_UTIL, alt, cfg, img or RC_KERNEL_BIN)
     assert find_device.called
-    assert req.called_with(exe)
+    assert req.call_args_list == [call(exe or DFU_UTIL)]
     assert cc.call_args_list == [call(EXPECTED_COMMAND[map_tc])]

